vital/types/metadata_types.cxx

```diff
@@ -18,4 +23,80 @@ operator<<( std::ostream& os, std_0102_lds const& )
   return os << "std_0102_local_set";
 }
 
+// ----------------------------------------------------------------------------
+uint64_t
+std_0104_datetime_to_unix_timestamp( std::string const& value )
+{
+  try
+  {
+    // Check datetime length
+    auto const length = std::string("YYYYMMDDThhmmss").size();
+    if( value.size() != length )
+    {
+      VITAL_THROW( metadata_exception, "invalid length" );
+    }
+
+    // Check datetime format
+    for ( size_t i = 0; i < length; ++i )
+    {
+      auto const c = value[ i ];
+      if( ( i == 8 && c != 'T' ) || ( i != 8 && !std::isdigit( c ) ) )
+      {
+          VITAL_THROW( metadata_exception, "invalid format" );
+      }
+    }
+
+    // Parse datetime fields
+    std::tm datetime = {};
+    datetime.tm_year =  std::stoi( value.substr( 0, 4 ) ) - 1900;
+    datetime.tm_mon =   std::stoi( value.substr( 4, 2 ) ) - 1;
+    datetime.tm_mday =  std::stoi( value.substr( 6, 2 ) );
+    datetime.tm_hour =  std::stoi( value.substr( 9, 2 ) );
+    datetime.tm_min =   std::stoi( value.substr( 11, 2 ) );
+    datetime.tm_sec =   std::stoi( value.substr( 13, 2 ) );
+    datetime.tm_wday =  -1;
+    datetime.tm_yday =  -1;
+    datetime.tm_isdst = -1;
+
+    auto const record = datetime;
+
+    // Call system function to convert UTC -> Unix
+    std::time_t const timestamp =
+#ifdef _WIN32
+      _mkgmtime64( &datetime );
+#else
+      timegm( &datetime );
+#endif
+
+    // Struct will be corrected if given date is invalid (e.g. hour = 24 ->
+    // hour = 0 and day += 1). Catch when this happens and throw, since our
+    // input is expected to be valid
+    if( datetime.tm_year != record.tm_year ||
+        datetime.tm_mon  != record.tm_mon ||
+        datetime.tm_mday != record.tm_mday ||
+        datetime.tm_hour != record.tm_hour ||
+        datetime.tm_min  != record.tm_min ||
+        datetime.tm_sec  != record.tm_sec )
+    {
+      VITAL_THROW( metadata_exception, "invalid datetime" );
+    }
+
+    // (time_t)(-1) is returned if date is out of representable range.
+    // If time_t is signed, negative values are possible (before epoch), but
+    // not representable by our uint64_t return type
+    if( timestamp == static_cast< std::time_t >( -1 ) || timestamp < 0 )
```

Comment:
This is probably going to generate an annoying warning if `time_t` is unsigned. Unfortunately, I don't know any good way to suppress that without `if constexpr` (C++17).

Since 1969-12-31T23:59:59 UTC is, however, a perfectly valid date, I'd be tempted to add a note to fix this once C++17 is available. The code probably should look like:
```c++
if constexpr( std::is_signed_v< time_t > )
{
  if( timestamp < 0 ) // throw ...
}
else
{
  if( timestamp == static_cast< std::time_t >( -1 ) ) // throw ...
}
```

(Strictly speaking, one would need to check `errno` if an actual `-1` is returned, but since we can't represent that anyway, doing so is sort of a moot point.)

Of course, you _could_ do the above wrapped in `#if __cplusplus >= 201703L`...